app/src/main/java/com/github/st0rmtroop3r/weather/view_model/MainViewModel.kt

```diff
@@ -1,7 +1,46 @@
-package com.github.st0rmtroop3r.weather.ui.main
+package com.github.st0rmtroop3r.weather.view_model
 
-import android.arch.lifecycle.ViewModel
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.github.st0rmtroop3r.weather.WeatherApp
+import com.github.st0rmtroop3r.weather.model.entities.WeatherResponse
+import com.github.st0rmtroop3r.weather.model.network.OpenWeatherMapApi
+import kotlinx.coroutines.runBlocking
+import javax.inject.Inject
 
 class MainViewModel : ViewModel() {
-    // TODO: Implement the ViewModel
+
+    private val TAG = this.javaClass.simpleName
+
+    @Inject
+    lateinit var api : OpenWeatherMapApi
+
+    private val currentWeather = MutableLiveData<WeatherResponse>()
+    private val currentWeatherError = MutableLiveData<String>()
+
+    init {
+        WeatherApp.appComponent.inject(this)
```

Comment:
I'd suggest you reading this
https://proandroiddev.com/viewmodel-with-dagger2-architecture-components-2e06f06c9455

Using constructor injection is a better approach. Also, you could try using AndroidInject